### PR TITLE
Add NRIC checker and assertions

### PR DESCRIPTION
**Pull request by hiongkaihan**

### CODE REVIEW

File: src/main/java/seedu/duke/command/ListCommand.java [hunk 1]
```diff
@@ -30,10 +30,10 @@ public void execute() throws Exception {
             list += "\n" + ++patientCount + ". " + patientID;
         }
 
-        // Todo implement more exceptions
         if (patientCount == 0) {
             throw new Exception(Constants.EXCEPTION_LIST_EMPTY);
         } else {
+            assert list != null : "List should not be empty";
```
Comment (brandonfoong) [inline src/main/java/seedu/duke/command/ListCommand.java:36]: You could change "not be empty" to "not be null" to reduce confusion between list being `null` or an empty string "".

File: src/main/java/seedu/duke/command/AddCommand.java [hunk 2]
```diff
@@ -21,17 +21,44 @@ public AddCommand(Ui ui, Data data, HashMap<String, String> arguments) {
...
-        // TODO Replace by ui after ui is implemented
+
         String patientID = arguments.get("payload");
+        patientID = patientID.toUpperCase();
         int stringLength = patientID.length();
+        boolean validID = true;
 
-        // TODO more ways to check for invalid NRIC and other exceptions
+        // Checks if ID has 9 characters
         if (stringLength != 9) {
```
Comment (brandonfoong) [inline src/main/java/seedu/duke/command/AddCommand.java:31]: Would it be better to extract this as a global constant to avoid using magic values?

**Review by brandonfoong — COMMENTED** [review]
Looks good, but I'll just leave this open (for now) for others to check.

File: src/main/java/seedu/duke/command/ListCommand.java (hunk 1 above)
Comment (hiongkaihan) [reply src/main/java/seedu/duke/command/ListCommand.java]: I have done the change to ""

File: src/main/java/seedu/duke/command/AddCommand.java (hunk 2 above)
Comment (hiongkaihan) [reply src/main/java/seedu/duke/command/AddCommand.java]: I have extracted all the magic values and used global constants instead

**Pull request merged**